metricflow/instances.py

```diff
@@ -22,61 +22,6 @@
 )
 
 
-class ModelReference(SerializableDataclass):
```

Comment:
@QMalcolm @WilliamDee - do we want to move these? As far as I know they're primarily used inside of our internal semantic containers. The DataSourceReference, for example, is only used as a typehint for the property allowing us to access the reference object. If that's the case, it might make sense to keep these in MetricFlow, and have our internal model object implementations (whatever they may look like) add those properties.

If we have some specific use for these in mind - or if we just prefer to move them because then we can have the interfaces package provide a functioning suite of model objects for ease of testing parsing behavior and so forth - that's perfectly fine. Just thought I'd ask before we get too far!

Reply:
So if we only look at what is in dbt_semantic_interfaces right now, references are not necessary. However, talking to @plypaul earlier today, it sounds like we'll need either References or Specs in dbt_semantic_interfaces for resolving `{{ dimension('dim_name') }}` and etc in dbt_semantic_interfaces. When weighing Specs and References, references seemed the easier way to go in that we end up moving less to dbt_semantic_interfaces and it is less tangled.

Reply:
Ah yes, that makes perfect sense. I think resolving to a Reference is the right choice.